python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -902,6 +904,18 @@ def freshness_policies_by_key(self) -> Mapping[AssetKey, FreshnessPolicy]:
     def auto_materialize_policies_by_key(self) -> Mapping[AssetKey, AutoMaterializePolicy]:
         return self._auto_materialize_policies_by_key
 
+    # Applies only to external observable assets. Can be removed when we fold
+    # `auto_observe_interval_minutes` into auto-materialize policies.
+    @property
+    def auto_observe_interval_minutes(self) -> Optional[float]:
+        first_key = next(iter(self.keys), None)
+        if first_key:
+            return self.metadata_by_key.get(first_key, {}).get(
+                SYSTEM_METADATA_KEY_AUTO_OBSERVE_INTERVAL_MINUTES
+            )
+        else:
+            return None
+
```

Comment:
this can share logic with your upstream helper function if you parameterize the metadata key.

Also would be good to do a runtime invariant of the type coming out of the dict since I don't think that is truly type safe?